torch/testing/_asserts.py

```diff
@@ -148,6 +165,13 @@ def _check_attributes_equal(
     if check_device and actual.device != expected.device:
         return _TestingErrorMeta(AssertionError, msg_fmtstr.format("device", actual.device, expected.device))
 
+    if actual.is_quantized != expected.is_quantized:
+        return _TestingErrorMeta(
+            AssertionError, msg_fmtstr.format("is_quantized", actual.is_quantized, expected.is_quantized)
+        )
+    elif actual.is_quantized and actual.qscheme() != expected.qscheme():
```

Comment:
Is there a use case to disable this check? Since the actual value checking is now agnostic to the quantization scheme, we could also add a `check_qscheme: bool = True` kwarg that would allow tensors of different schemes to be compared.

Reply:
I'm not sure if we'll have these use cases in the future but it can be achieved by dequantizing both Tensors and then compare as well. For now we can just compare qscheme and dtype I think